From 7c661c4d73b6ae58ec6cdfbafce080006be2ae7e Mon Sep 17 00:00:00 2001
From: Jennifer Lee <jennifer1.lee@intel.com>
Date: Thu, 14 Feb 2019 14:54:45 -0800
Subject: [PATCH] Modify the ID of software image updater object on DBus to

 allow force update onto same version image

In the original design of image update, it does not allow the same version of image to be flashed onto itself.
But this blocks validation tests and in most of the cases we don't prevent user from doing such update.

This patch appends a random number after the version ID hash string to unblock such limitation.

Signed-off-by: Jennifer Lee <jennifer1.lee@intel.com>
Change-Id: I16aba4804ae1bc2e8784320f91c0419fb8b23c35

---
 image_manager.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/image_manager.cpp b/image_manager.cpp
index eb4fec4..9b66961 100644
--- a/image_manager.cpp
+++ b/image_manager.cpp
@@ -9,6 +9,7 @@
 #include <stdlib.h>
 #include <sys/stat.h>
 #include <sys/wait.h>
+#include <time.h>
 #include <unistd.h>
 
 #include <elog-errors.hpp>
@@ -196,6 +197,11 @@ int Manager::processImage(const std::string& tarFilePath)
     auto salt = std::to_string(randomGen());
     auto id = Version::getId(version + salt);
 
+    // Append a random number after the original version hash
+    // This will allow forcing image update onto the same version
+    srand(time(NULL));
+    id = id + "_" + std::to_string(rand());
+
     fs::path imageDirPath = std::string{IMG_UPLOAD_DIR};
     imageDirPath /= id;
 
